util/log/log.go

```diff
@@ -34,74 +43,169 @@ func FatalOnPanic() {
 	}
 }
 
+func logKV(buf *bytes.Buffer, kvs ...[]interface{}) {
+	var i int
+	var s string
+	var kv []interface{}
+	for kvi := range kvs {
+		kv = kvs[kvi]
+		l := len(kv)
+		for i = 0; i < l; i++ {
+			if i%2 == 0 {
+				if kvi != 0 || i != 0 {
+					buf.WriteString(" ")
+				}
+			} else {
+				buf.WriteString("=")
+			}
+			switch kt := kv[i].(type) {
+			case string:
+				s = kt
+			case fmt.Stringer:
+				s = kt.String()
+			default:
+				s = fmt.Sprintf("%v", kt)
+			}
+			if i%2 == 0 {
+				buf.WriteString(s)
+			} else {
+				buf.WriteString(strconv.Quote(s))
+			}
+		}
+		if i%2 != 0 {
+			panic(fmt.Sprintf("odd number of key-value pairs passed: %v", kv))
+		}
+	}
+	buf.WriteString("\n")
+}
+
+func headerKV(sev clog.Severity, depth int, msg string) []interface{} {
+	ret := make([]interface{}, 10)
+	ret[0], ret[1] = "level", clog.SeverityName[sev]
+	ret[2], ret[3] = "time", time.Now().Format(time.RFC3339)
+	ret[4], ret[6] = "file", "line"
```

Comment:
This... "warp" here is too confusing.  I think that two variables could be spared for the output of clog.Caller.
```
file, line := clog.Caller(depth + 1)
return []interface{}{
  "level", clog.SeverityName[sev],
  "time", time.Now().Format(time.RFC3339)
  "file", file,
  "line", line,
  "msg", msg,
}
```